tests/test_join.py

```diff
@@ -11,134 +11,130 @@ def t1(db: gp.Database):
 
-def test_join_both_all_targets(db: gp.Database, t1: gp.Table, t2: gp.Table):
-    ret = t1._join(
-        t2,
-        targets=[t1["*"], t2["*"]],
-        how="INNER JOIN",
-        on_str="ON temp1.id1 = temp2.id2",
-    ).fetch()
-    assert list(list(ret)[0].keys()) == ["id1", "idd1", "n1", "id2", "idd2", "n2"]
+def test_join_all_and_all_columns(db: gp.Database, t1: gp.Table, t2: gp.Table):
+    row = next(
+        iter(
+            t1.join(
+                t2, cond=t1["id1"] == t2["id2"], self_columns={"*"}, other_columns={"*"}
+            ).fetch()
+        )
+    )
+    for col in ["id1", "idd1", "n1", "id2", "idd2", "n2"]:
+        assert col in row
 
 
-def test_join_both_empty_targets(db: gp.Database, t1: gp.Table, t2: gp.Table):
-    ret = t1._join(t2, targets=[], how="INNER JOIN", on_str="ON temp1.id1 = temp2.id2").fetch()
-    assert list(list(ret)[0].keys()) == ["id1", "idd1", "n1", "id2", "idd2", "n2"]
+def test_join_no_and_no_columns(db: gp.Database, t1: gp.Table, t2: gp.Table):
+    ret = next(iter(t1.join(t2, cond=t1["id1"] == t2["id2"]).fetch()))
+    assert "id1" not in ret and "id2" not in ret  # NOTE: The row is not empty with psycopg2
 
 
-def test_join_all_empty_targets(db: gp.Database, t1: gp.Table, t2: gp.Table):
-    ret = t1._join(
-        t2, targets=[t2["*"]], how="INNER JOIN", on_str="ON temp1.id1 = temp2.id2"
-    ).fetch()
-    assert list(list(ret)[0].keys()) == ["id2", "idd2", "n2"]
+def test_join_all_and_no_columns(db: gp.Database, t1: gp.Table, t2: gp.Table):
+    ret = next(iter(t1.join(t2, cond=t1["id1"] == t2["id2"], other_columns={"*"}).fetch()))
+    for col in ["id2", "idd2", "n2"]:
+        assert col in ret
 
 
-def test_join_all_one_targets(db: gp.Database, t1: gp.Table, t2: gp.Table):
-    ret = t1._join(
-        t2, targets=[t2["*"], t1["id1"]], how="INNER JOIN", on_str="ON temp1.id1 = temp2.id2"
+def test_join_all_and_one_columns(db: gp.Database, t1: gp.Table, t2: gp.Table):
+    ret = next(
+        iter(
+            t1.join(
+                t2, cond=t1["id1"] == t2["id2"], self_columns={"id1"}, other_columns={"*"}
+            ).fetch()
+        )
     )
-    assert list(list(ret.fetch())[0].keys()) == ["id2", "idd2", "n2", "id1"]
-    assert ret.columns is None
-
-
-def test_join_both_mulp_targets(db: gp.Database, t1: gp.Table, t2: gp.Table):
-    ret = t1._join(
-        t2,
-        targets=[t1["id1"], t1["n1"], t2["idd2"]],
-        how="INNER JOIN",
-        on_str="ON temp1.id1 = temp2.id2",
-    ).fetch()
-    assert list(list(ret)[0].keys()) == ["id1", "n1", "idd2"]
+    for col in ["id2", "idd2", "n2", "id1"]:
+        assert col in ret
+
+
+def test_join_columns_from_list(db: gp.Database, t1: gp.Table, t2: gp.Table):
+    ret = next(
+        iter(
+            t1.join(
+                t2,
+                cond=t1["id1"] == t2["id2"],
+                self_columns=dict.fromkeys(["id1", "n1"]),
+                other_columns={"idd2"},
+            ).fetch()
+        )
+    )
```

Comment:
there are quite some  next(iter(....)) can we make it to some property like 
 t1.join(
                t2,
                cond=t1["id1"] == t2["id2"],
                self_columns=dict.fromkeys(["id1", "n1"]),
                other_columns={"idd2"},
            ).fetch().row?